### PR TITLE
Fix aks multi-environment  support

### PR DESCRIPTION
**Pull request by smallteeths**
https://github.com/rancher/rancher/issues/38225

### CODE REVIEW

File: pkg/aks/client.go [hunk 1]
```diff
@@ -186,3 +188,18 @@ func GetCachedTenantID(secretClient secretClient, subscriptionID string, secret
 	}
 	return tenantID, err
 }
+
+func GetEnvironment(env string) azure.Environment {
+	switch env {
+	case "AzurePublicCloud":
+		return azure.PublicCloud
```
Comment (Jason-ZW) [inline pkg/aks/client.go:195]: Useless `case` statement

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
Others LGTM

**Comment by smallteeths** [comment]
> Should also update the README.md for the new secret data entry in https://github.com/rancher/aks-operator/blob/master/README.md#5-create-aks-secret-with-clientid-and-clientsecret.

PR updated.

**Review by Jason-ZW — APPROVED** [review]

**Review by Jason-ZW — APPROVED** [review]
LGTM

**Comment by mjura** [comment]
Could you please rebase ?

**Comment by mjura** [comment]
It will be reworked in https://github.com/rancher/aks-operator/pull/136

**Pull request closed without merging**